stm32_base/hal/imu_stm32/invensense_9150.c

```diff
@@ -35,13 +58,155 @@ static int is9150_read_reg_block(uint8_t start_addr, size_t len,
     return res;
 }
 
-int is9150_init(struct i2c_dev *dev, uint8_t addr)
+
+static int is9150_read_mag_reg_block(uint8_t start_addr, size_t len,
+                                     uint8_t *rx_buf)
+{
+    int res = 0;
```

Comment:
Pointless to initialize this if it will always be overwritten.  Applies throughout.